src/fs/file.rs

```diff
@@ -290,6 +291,56 @@ impl File {
         op.await
     }
 
+    /// Like [`read_at`], but using a pre-mapped buffer
+    /// registered with [`BufRegistry`].
```

Comment:
Should this comment point out a slice of the registered buffer is also allowed? The signature does make it obvious but I'm still confused myself whether the slicing has to be made from front to back or could an app start from the back, or even the middle if they wanted? Well, maybe not a question for this review.